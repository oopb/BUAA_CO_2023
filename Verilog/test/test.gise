<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<generated_project xmlns="http://www.xilinx.com/XMLSchema" xmlns:xil_pn="http://www.xilinx.com/XMLSchema">

  <!--                                                          -->

  <!--             For tool use only. Do not edit.              -->

  <!--                                                          -->

  <!-- ProjectNavigator created generated project file.         -->

  <!-- For use in tracking generated file and other information -->

  <!-- allowing preservation of process status.                 -->

  <!--                                                          -->

  <!-- Copyright (c) 1995-2013 Xilinx, Inc.  All rights reserved. -->

  <version xmlns="http://www.xilinx.com/XMLSchema">11.1</version>

  <sourceproject xmlns="http://www.xilinx.com/XMLSchema" xil_pn:fileType="FILE_XISE" xil_pn:name="test.xise"/>

  <files xmlns="http://www.xilinx.com/XMLSchema">
    <file xil_pn:fileType="FILE_DIRECTORY" xil_pn:name="_ngo"/>
    <file xil_pn:fileType="FILE_XMSGS" xil_pn:name="_xmsgs/ngdbuild.xmsgs"/>
    <file xil_pn:fileType="FILE_XMSGS" xil_pn:name="_xmsgs/xst.xmsgs"/>
    <file xil_pn:branch="Implementation" xil_pn:fileType="FILE_NGDBUILD_LOG" xil_pn:name="counting_tb.bld"/>
    <file xil_pn:fileType="FILE_CMD_LOG" xil_pn:name="counting_tb.cmd_log"/>
    <file xil_pn:branch="Implementation" xil_pn:fileType="FILE_LSO" xil_pn:name="counting_tb.lso"/>
    <file xil_pn:branch="Implementation" xil_pn:fileType="FILE_NGC" xil_pn:name="counting_tb.ngc"/>
    <file xil_pn:branch="Implementation" xil_pn:fileType="FILE_NGR" xil_pn:name="counting_tb.ngr"/>
    <file xil_pn:branch="Implementation" xil_pn:fileType="FILE_XST_PROJECT" xil_pn:name="counting_tb.prj"/>
    <file xil_pn:branch="Implementation" xil_pn:fileType="FILE_XST_STX" xil_pn:name="counting_tb.stx"/>
    <file xil_pn:branch="Implementation" xil_pn:fileType="FILE_XST_REPORT" xil_pn:name="counting_tb.syr"/>
    <file xil_pn:branch="Implementation" xil_pn:fileType="FILE_XST" xil_pn:name="counting_tb.xst"/>
    <file xil_pn:fileType="FILE_XST_PROJECT" xil_pn:name="counting_tb_beh.prj"/>
    <file xil_pn:branch="BehavioralSim" xil_pn:fileType="FILE_ISIM_EXE" xil_pn:name="counting_tb_isim_beh.exe"/>
    <file xil_pn:fileType="FILE_ISIM_MISC" xil_pn:name="counting_tb_isim_beh.wdb"/>
    <file xil_pn:fileType="FILE_XRPT" xil_pn:name="counting_tb_ngdbuild.xrpt"/>
    <file xil_pn:fileType="FILE_XST_PROJECT" xil_pn:name="counting_tb_stx_beh.prj"/>
    <file xil_pn:fileType="FILE_XRPT" xil_pn:name="counting_tb_xst.xrpt"/>
    <file xil_pn:fileType="FILE_LOG" xil_pn:name="fuse.log"/>
    <file xil_pn:fileType="FILE_DIRECTORY" xil_pn:name="isim"/>
    <file xil_pn:branch="BehavioralSim" xil_pn:fileType="FILE_CMD" xil_pn:name="isim.cmd"/>
    <file xil_pn:branch="BehavioralSim" xil_pn:fileType="FILE_LOG" xil_pn:name="isim.log"/>
    <file xil_pn:fileType="FILE_FITTER_REPORT" xil_pn:name="webtalk_pn.xml"/>
    <file xil_pn:branch="BehavioralSim" xil_pn:fileType="FILE_INI" xil_pn:name="xilinxsim.ini"/>
    <file xil_pn:fileType="FILE_DIRECTORY" xil_pn:name="xlnx_auto_0_xdb"/>
    <file xil_pn:fileType="FILE_DIRECTORY" xil_pn:name="xst"/>
  </files>

  <transforms xmlns="http://www.xilinx.com/XMLSchema">
    <transform xil_pn:end_ts="1694694853" xil_pn:name="TRAN_copyInitialToAbstractSimulation" xil_pn:start_ts="1694694853">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694853" xil_pn:in_ck="-8753906528066462187" xil_pn:name="TRAN_copyAbstractToPostAbstractSimulation" xil_pn:start_ts="1694694853">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
      <outfile xil_pn:name="ALU.v"/>
      <outfile xil_pn:name="counting.v"/>
      <outfile xil_pn:name="test.v"/>
    </transform>
    <transform xil_pn:end_ts="1694694859" xil_pn:name="TRAN_xawsToSimhdl" xil_pn:prop_ck="-9114130925493496045" xil_pn:start_ts="1694694859">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694859" xil_pn:name="TRAN_schematicsToHdlSim" xil_pn:prop_ck="4072996776535561649" xil_pn:start_ts="1694694859">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694859" xil_pn:name="TRAN_regenerateCoresSim" xil_pn:prop_ck="-2445722113884837223" xil_pn:start_ts="1694694859">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694859" xil_pn:in_ck="-8753906528066462187" xil_pn:name="TRAN_copyPostAbstractToPreSimulation" xil_pn:start_ts="1694694859">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
      <outfile xil_pn:name="ALU.v"/>
      <outfile xil_pn:name="counting.v"/>
      <outfile xil_pn:name="test.v"/>
    </transform>
    <transform xil_pn:end_ts="1694694865" xil_pn:in_ck="-8753906528066462187" xil_pn:name="TRAN_ISimulateBehavioralModelRunFuse" xil_pn:prop_ck="-9075756749317109771" xil_pn:start_ts="1694694859">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
      <outfile xil_pn:name="counting_tb_beh.prj"/>
      <outfile xil_pn:name="counting_tb_isim_beh.exe"/>
      <outfile xil_pn:name="fuse.log"/>
      <outfile xil_pn:name="isim"/>
      <outfile xil_pn:name="xilinxsim.ini"/>
    </transform>
    <transform xil_pn:end_ts="1694695018" xil_pn:in_ck="2135877875325013131" xil_pn:name="TRAN_ISimulateBehavioralModel" xil_pn:prop_ck="-3773256309119865576" xil_pn:start_ts="1694695018">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
      <status xil_pn:value="OutOfDateForOutputs"/>
      <status xil_pn:value="OutputChanged"/>
      <outfile xil_pn:name="counting_tb_isim_beh.wdb"/>
      <outfile xil_pn:name="isim.cmd"/>
      <outfile xil_pn:name="isim.log"/>
    </transform>
    <transform xil_pn:end_ts="1694694314" xil_pn:name="TRAN_copyInitialToXSTAbstractSynthesis" xil_pn:start_ts="1694694314">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694314" xil_pn:name="TRAN_schematicsToHdl" xil_pn:prop_ck="-4745201726441216975" xil_pn:start_ts="1694694314">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694314" xil_pn:name="TRAN_regenerateCores" xil_pn:prop_ck="-2445722113884837223" xil_pn:start_ts="1694694314">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694314" xil_pn:name="TRAN_SubProjectAbstractToPreProxy" xil_pn:start_ts="1694694314">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694314" xil_pn:name="TRAN_xawsTohdl" xil_pn:prop_ck="-9114130925493496045" xil_pn:start_ts="1694694314">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694314" xil_pn:name="TRAN_SubProjectPreToStructuralProxy" xil_pn:prop_ck="3206529612922900429" xil_pn:start_ts="1694694314">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694314" xil_pn:name="TRAN_platgen" xil_pn:prop_ck="6340243462521710819" xil_pn:start_ts="1694694314">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694341" xil_pn:in_ck="-8753906528066462187" xil_pn:name="TRANEXT_xstsynthesize_virtex6" xil_pn:prop_ck="-1994276729370490442" xil_pn:start_ts="1694694314">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="WarningsGenerated"/>
      <status xil_pn:value="ReadyToRun"/>
      <outfile xil_pn:name="_xmsgs/xst.xmsgs"/>
      <outfile xil_pn:name="counting_tb.lso"/>
      <outfile xil_pn:name="counting_tb.ngc"/>
      <outfile xil_pn:name="counting_tb.ngr"/>
      <outfile xil_pn:name="counting_tb.prj"/>
      <outfile xil_pn:name="counting_tb.stx"/>
      <outfile xil_pn:name="counting_tb.syr"/>
      <outfile xil_pn:name="counting_tb.xst"/>
      <outfile xil_pn:name="counting_tb_xst.xrpt"/>
      <outfile xil_pn:name="webtalk_pn.xml"/>
      <outfile xil_pn:name="xst"/>
    </transform>
    <transform xil_pn:end_ts="1694694341" xil_pn:name="TRAN_compileBCD2" xil_pn:prop_ck="-5830255872031621826" xil_pn:start_ts="1694694341">
      <status xil_pn:value="SuccessfullyRun"/>
      <status xil_pn:value="ReadyToRun"/>
    </transform>
    <transform xil_pn:end_ts="1694694354" xil_pn:in_ck="-446513304173399038" xil_pn:name="TRANEXT_ngdbuild_FPGA" xil_pn:prop_ck="-1525529507562365590" xil_pn:start_ts="1694694341">
      <status xil_pn:value="FailedRun"/>
      <status xil_pn:value="WarningsGenerated"/>
      <status xil_pn:value="ReadyToRun"/>
      <outfile xil_pn:name="_ngo"/>
      <outfile xil_pn:name="_xmsgs/ngdbuild.xmsgs"/>
      <outfile xil_pn:name="counting_tb.bld"/>
      <outfile xil_pn:name="counting_tb_ngdbuild.xrpt"/>
    </transform>
  </transforms>

</generated_project>
