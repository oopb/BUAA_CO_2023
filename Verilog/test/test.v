`timescale 1ns / 1ps
`include "counting.v"

module counting_tb;
    reg  [1:0] num;
    reg        clk;
    wire       ans;
    counting counting_inst (
        .num(num),
        .clk(clk),
        .ans(ans)
    );

    always #5 clk = !clk;

    initial begin
        //$dumpfile("counting.vcd");
        //$dumpvars;
        #5 num = 2'd1;
        #5 num = 2'd3;
        #5 num = 2'd1;
        #5 num = 2'd2;
        #5 num = 2'd2;
        #5 num = 2'd3;
        #5 num = 2'd2;
    end


endmodule
