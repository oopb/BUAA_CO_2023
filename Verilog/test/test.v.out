#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1110-g18392a46)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "D:\iverilog\lib\ivl\system.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "D:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "D:\iverilog\lib\ivl\va_math.vpi";
S_000001af07c3a7d0 .scope module, "counting_tb" "counting_tb" 2 4;
 .timescale -9 -12;
v000001af07d448e0_0 .net "ans", 0 0, L_000001af07cf31e0;  1 drivers
v000001af07cf33c0_0 .var "clk", 0 0;
v000001af07cf30a0_0 .var "num", 1 0;
S_000001af07c3ba00 .scope module, "counting_inst" "counting" 2 8, 3 27 0, S_000001af07c3a7d0;
 .timescale -9 -12;
    .port_info 0 /INPUT 2 "num";
    .port_info 1 /INPUT 1 "clk";
    .port_info 2 /OUTPUT 1 "ans";
L_000001af07d44988 .functor BUFT 1, C4<11>, C4<0>, C4<0>, C4<0>;
v000001af07c36d00_0 .net/2u *"_ivl_0", 1 0, L_000001af07d44988;  1 drivers
v000001af07c36ae0_0 .net *"_ivl_2", 0 0, L_000001af07cf3140;  1 drivers
L_000001af07d449d0 .functor BUFT 1, C4<01>, C4<0>, C4<0>, C4<0>;
v000001af07cfc050_0 .net/2u *"_ivl_4", 1 0, L_000001af07d449d0;  1 drivers
L_000001af07d44a18 .functor BUFT 1, C4<00>, C4<0>, C4<0>, C4<0>;
v000001af07c366c0_0 .net/2u *"_ivl_6", 1 0, L_000001af07d44a18;  1 drivers
v000001af07c3bb90_0 .net *"_ivl_8", 1 0, L_000001af07cf3640;  1 drivers
v000001af07c3bc30_0 .net "ans", 0 0, L_000001af07cf31e0;  alias, 1 drivers
v000001af07c3bcd0_0 .net "clk", 0 0, v000001af07cf33c0_0;  1 drivers
v000001af07c3bd70_0 .net "num", 1 0, v000001af07cf30a0_0;  1 drivers
v000001af07d44840_0 .var "status", 1 0;
E_000001af07c39b70 .event posedge, v000001af07c3bcd0_0;
L_000001af07cf3140 .cmp/eq 2, v000001af07d44840_0, L_000001af07d44988;
L_000001af07cf3640 .functor MUXZ 2, L_000001af07d44a18, L_000001af07d449d0, L_000001af07cf3140, C4<>;
L_000001af07cf31e0 .part L_000001af07cf3640, 0, 1;
    .scope S_000001af07c3ba00;
T_0 ;
    %pushi/vec4 0, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %end;
    .thread T_0;
    .scope S_000001af07c3ba00;
T_1 ;
    %wait E_000001af07c39b70;
    %load/vec4 v000001af07d44840_0;
    %dup/vec4;
    %pushi/vec4 0, 0, 2;
    %cmp/u;
    %jmp/1 T_1.0, 6;
    %dup/vec4;
    %pushi/vec4 1, 0, 2;
    %cmp/u;
    %jmp/1 T_1.1, 6;
    %dup/vec4;
    %pushi/vec4 2, 0, 2;
    %cmp/u;
    %jmp/1 T_1.2, 6;
    %dup/vec4;
    %pushi/vec4 3, 0, 2;
    %cmp/u;
    %jmp/1 T_1.3, 6;
    %jmp T_1.4;
T_1.0 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 1, 0, 2;
    %jmp/0xz  T_1.5, 4;
    %pushi/vec4 1, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.6;
T_1.5 ;
    %pushi/vec4 0, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
T_1.6 ;
    %jmp T_1.4;
T_1.1 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 1, 0, 2;
    %jmp/0xz  T_1.7, 4;
    %pushi/vec4 1, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.8;
T_1.7 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 2, 0, 2;
    %jmp/0xz  T_1.9, 4;
    %pushi/vec4 2, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.10;
T_1.9 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 3, 0, 2;
    %jmp/0xz  T_1.11, 4;
    %pushi/vec4 0, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.12;
T_1.11 ;
    %pushi/vec4 0, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
T_1.12 ;
T_1.10 ;
T_1.8 ;
    %jmp T_1.4;
T_1.2 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 1, 0, 2;
    %jmp/0xz  T_1.13, 4;
    %pushi/vec4 1, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.14;
T_1.13 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 2, 0, 2;
    %jmp/0xz  T_1.15, 4;
    %pushi/vec4 2, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.16;
T_1.15 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 3, 0, 2;
    %jmp/0xz  T_1.17, 4;
    %pushi/vec4 3, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.18;
T_1.17 ;
    %pushi/vec4 0, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
T_1.18 ;
T_1.16 ;
T_1.14 ;
    %jmp T_1.4;
T_1.3 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 1, 0, 2;
    %jmp/0xz  T_1.19, 4;
    %pushi/vec4 1, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.20;
T_1.19 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 2, 0, 2;
    %jmp/0xz  T_1.21, 4;
    %pushi/vec4 0, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.22;
T_1.21 ;
    %load/vec4 v000001af07c3bd70_0;
    %cmpi/e 3, 0, 2;
    %jmp/0xz  T_1.23, 4;
    %pushi/vec4 3, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
    %jmp T_1.24;
T_1.23 ;
    %pushi/vec4 0, 0, 2;
    %assign/vec4 v000001af07d44840_0, 0;
T_1.24 ;
T_1.22 ;
T_1.20 ;
    %jmp T_1.4;
T_1.4 ;
    %pop/vec4 1;
    %jmp T_1;
    .thread T_1;
    .scope S_000001af07c3a7d0;
T_2 ;
    %delay 5000, 0;
    %load/vec4 v000001af07cf33c0_0;
    %nor/r;
    %store/vec4 v000001af07cf33c0_0, 0, 1;
    %jmp T_2;
    .thread T_2;
    .scope S_000001af07c3a7d0;
T_3 ;
    %delay 5000, 0;
    %pushi/vec4 1, 0, 2;
    %store/vec4 v000001af07cf30a0_0, 0, 2;
    %delay 5000, 0;
    %pushi/vec4 3, 0, 2;
    %store/vec4 v000001af07cf30a0_0, 0, 2;
    %delay 5000, 0;
    %pushi/vec4 1, 0, 2;
    %store/vec4 v000001af07cf30a0_0, 0, 2;
    %delay 5000, 0;
    %pushi/vec4 2, 0, 2;
    %store/vec4 v000001af07cf30a0_0, 0, 2;
    %delay 5000, 0;
    %pushi/vec4 2, 0, 2;
    %store/vec4 v000001af07cf30a0_0, 0, 2;
    %delay 5000, 0;
    %pushi/vec4 3, 0, 2;
    %store/vec4 v000001af07cf30a0_0, 0, 2;
    %delay 5000, 0;
    %pushi/vec4 2, 0, 2;
    %store/vec4 v000001af07cf30a0_0, 0, 2;
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "test.v";
    "./counting.v";
